File: Homework/Ngan_Luu/346-moving-average-from-data-stream.py
https://leetcode.com/problems/moving-average-from-data-stream/
class MovingAverage:

    def __init__(self, size: int):
        self.size=size
        self.Queue=[]

    def next(self, val: int) -> float:
        if len(self.Queue)>=self.size:
            self.Queue.pop(0)
        self.Queue.append(val)
        return float(sum(self.Queue)/len(self.Queue))

File: Homework/Son/W2/346-moving-average-from-data-stream.js

class MovingAverage {
    constructor (size) {
        this._size = size
        this._values = []
    }

    next (val) {
        this._values.push(val)

        const sampleSet = this._values.slice(-this._size)
        const average = sampleSet.reduce(sum) / Math.min(sampleSet.length, this._size)

        return average
    }
}

function sum (a, b) {
    return a + b
}
